### PR TITLE
✨ Update readme

### PR DESCRIPTION
**Pull request by laurentsimon**
@olivekl can you advise?

closes https://github.com/gossts/slsa-go/issues/6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,166 @@
-# slsa-go
\ No newline at end of file
+# Generation of SLSA3+ provenance for golang binaries
+This repository contains a reference implementation for generating non-falsifiable [SLSA provenance](https://slsa.dev/) that meets the requirement for the [SLSA level 3 and above](https://slsa.dev/spec/v0.1/levels) for projects using the golang programming language.
```
Comment (asraa) [inline README.md:2]: Seems like most people will use "Go programming language" 
s/golang/Go

File: README.md [hunk 2]
```diff
@@ -1 +1,166 @@
...
+```
+
+### Workflow inputs
+
+The builder workflow [gossts/slsa-go/.github/workflows/builder.yml](.github/workflows/builder.yml) accepts the following inputs:
+
+| Name | Required | Description |
+| ------------ | -------- | ----------- |
+| `go-version` | no | The go version for your project. This value is passed, unchanged, to the [actions/setup-go](https://github.com/actions/setup-go) action when setting up the environment |
+| `env` | no | A list of environment variables, seperated by `,`: `VAR1: value, VAR2: value`. This is typically used to pass dynamically-generated values, such as `ldflags`.|
```
Comment (asraa) [inline README.md:60]: Let's mention any qualifications on the supported env vars?

File: README.md [hunk 3]
```diff
@@ -1 +1,166 @@
...
+[Verification of provenance](#verification)
+- [Inputs](#inputs)
+- [Command line examples](#command-line-example)
+
+[Technical design](#technial-design)
+- [Blog posts](#blog-posts)
+- [Specifications](#specifications)
+________
+
+## Generation
```
Comment (asraa) [inline README.md:21]: Suggest: add a subheader here for `Example provenance` even if left as a TODO

**Review by asraa — APPROVED** [review]
Can we ship this early and iterate?

File: README.md (hunk 2 above)
Comment (laurentsimon) [reply README.md:60]: Added `Note that only environment variables with names starting with `CGO_` or `GO` are accepted`

File: README.md (hunk 3 above)
Comment (laurentsimon) [reply README.md:21]: done, thanks

File: README.md (hunk 1 above)
Comment (laurentsimon) [reply README.md:2]: done, thanks!

**Pull request merged**